access_seats.go

```diff
@@ -53,6 +53,42 @@ func (api *API) UpdateAccessUserSeat(ctx context.Context, rc *ResourceContainer,
 		rc.Identifier,
 	)
 
+	// this requests expects an array of params, but this method only accepts a single param
+	res, err := api.makeRequestContext(ctx, http.MethodPatch, uri, []UpdateAccessUserSeatParams{params})
+	if err != nil {
+		return []AccessUpdateAccessUserSeatResult{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
+	}
+
+	var updateAccessUserSeatResponse UpdateAccessUserSeatResponse
+	err = json.Unmarshal(res, &updateAccessUserSeatResponse)
+	if err != nil {
+		return []AccessUpdateAccessUserSeatResult{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return updateAccessUserSeatResponse.Result, nil
+}
+
+// UpdateAccessUsersSeats updates many Access User Seats.
+//
+// API documentation: https://developers.cloudflare.com/api/operations/zero-trust-seats-update-a-user-seat
+func (api *API) UpdateAccessUsersSeats(ctx context.Context, rc *ResourceContainer, params []UpdateAccessUserSeatParams) ([]AccessUpdateAccessUserSeatResult, error) {
```

Comment:
the `params` here should match the [conventions](https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md) so `UpdateAccessUsersSeatsParams` and be a dedicated struct.

Reply:
Ok!
Could it be a type alias like `type UpdateAccessUsersSeatsParams []UpdateAccessUserSeatParams` referencing the struct from the other request ? Or does it need to be a completely new struct ?

Its worth to mention that they are both calling the same API method, and the only reason to not change the original method is to not break compatibility.

But now that I think about it, a variadic input would also work without breaking compatibility and without needing to add a new function. 
It would allow the users to call it with only one `UpdateAccessUserSeatParams` entries (as they do now), with multiple entries, or even with an array.

Ex: 
```go
func (api *API) UpdateAccessUserSeat(ctx context.Context, rc *ResourceContainer, params ...UpdateAccessUserSeatParams) ([]AccessUpdateAccessUserSeatResult, error) {
```
It could be used like this:
```go
// single entry
result, err := client.UpdateAccessUserSeat(
    ctx, AccountIdentifier(), UpdateAccessUserSeatParams{}
    )

// AND
// multiple entries
Result, err := client.UpdateAccessUserSeat(
    ctx, AccountIdentifier(), UpdateAccessUserSeatParams{}, UpdateAccessUserSeatParams{}
    )

// AND
// array into variadic 
Result, err := client.UpdateAccessUserSeat(
    ctx, AccountIdentifier(), []UpdateAccessUserSeatParams{}{}...
    )
```

Tell me what do you think! Thanks

Reply:
separate struct please. we have it in place to allow adding fields without needing to have as many breaking changes and it allows the two methods to change independently.

we have conventions for (at most) 3 parameters in the method signatures to prevent really long methods.

Reply:
I believe its up the the conventions now!
Thanks !
Let me know if you need anything else.